### PR TITLE
The cleaned codes for dataengine and dataset

### PR DESCRIPTION
**Pull request by ah89**
Please Review this code and leave comments

### CODE REVIEW

**Review by j48zheng — DISMISSED** [review]

**Review by j48zheng — DISMISSED** [review]

File: holoclean/dataengine.py [hunk 1]
```diff
@@ -6,33 +6,33 @@
...
     to read the input files and output to MySQL database
     """
 
-    def __init__(self, holoEnv):
+    def __init__(self, holo_env):
         """
         The constructor for DataEngine class
+
         Parameters
         ----------
```
Comment (gmichalo) [inline holoclean/dataengine.py:18]: As we use :param to specify the parameters I do not think we need to have  also in the comments the word:
Parameters

File: holoclean/dataengine.py [hunk 2]
```diff
@@ -42,39 +42,88 @@ def __init__(self, holoEnv):
...
+        """
+        Creating jdbc url for connection to database
+
+        Parameters
+        ----------
+        No parameter
+
+        Returns
+        -------
+        :return: jdbc_url : String
```
Comment (gmichalo) [inline holoclean/dataengine.py:75]: The same comment as before, as we use :return to specify the return objects I do not think we need to have also in the comments the word:
 Returns

File: holoclean/dataset.py [hunk 3]
```diff
@@ -102,30 +98,88 @@ def __init__(self):
...
+        -------
+        :return: random_str : String
+             Generating a string of random numbers
+        """
+        random_str = str(random.random())[2:]
+        return random_str
 
     def print_id(self):
+        """
+        This method writes dataset id inside dataset_id.txt file and return that
```
Comment (thodrek) [inline holoclean/dataset.py]: returns that id.

**Review by thodrek — DISMISSED** [review]
Looks good to me just fix that typo.

File: holoclean/dataengine.py [hunk 4]
```diff
@@ -109,10 +168,24 @@ def _add_meta(self, table_name, table_schema, dataset):
             self.db_backend.execute(create_table)
             self.db_backend.execute(add_row)
 
-    def _table_column_to_dataframe(self, table_name,
-                                   columns_name_list, dataset):
+    def _table_column_to_dataframe(self, table_name, columns_name_list,
+                                   dataset):
         """
-        This method get table general name and return it as spark dataframe
+        This method gets table general name and returns a spark dataframe
```
Comment (JVonderwell) [inline holoclean/dataengine.py:175]: typo - dataframe that contains

File: holoclean/dataengine.py [hunk 5]
```diff
@@ -109,10 +168,24 @@ def _add_meta(self, table_name, table_schema, dataset):
...
+            contains the column of that table
+
+        Parameters
+        ----------
+        :param table_name: The name of table that we want some of its columns
+        :param columns_name_list: The list of columns that we want from
+                                table_name
+        :param dataset: The data set object to access to the correct project
```
Comment (JVonderwell) [inline holoclean/dataengine.py]: typo - 'access the' not 'access to the'

File: holoclean/dataengine.py [hunk 6]
```diff
@@ -109,10 +168,24 @@ def _add_meta(self, table_name, table_schema, dataset):
...
+                        tables
+
+        Returns
+        -------
+        :return: Dataframe
+                It returns a dataframe contains "columns_name_list" columns data
```
Comment (JVonderwell) [inline holoclean/dataengine.py]: typo: dataframe that contains

File: holoclean/dataengine.py [hunk 7]
```diff
@@ -154,39 +244,51 @@ def _dataframe_to_table(self, spec_table_name, dataframe, append=0):
...
-        """
-        execute a query and create a dataframe from the results
+    def _query_spark(self, sql_query):
         """
+        Executing a query on the MySQL and create a dataframe from the results
 
+        Parameters
+        ----------
+        :param sql_query: String
+                    The query that we want to have it results on the dataframe
```
Comment (JVonderwell) [inline holoclean/dataengine.py]: wording: something as simple as 'query to execute' is probably fine here. it's a little wordy now.

File: holoclean/dataengine.py [hunk 8]
```diff
@@ -199,35 +301,47 @@ def get_schema(self, dataset, table_general_name):
...
+        """
+        This method returns MySQL database
+        Parameters
+        ----------
+        No parameter
+
+        Returns
+        -------
+        :return: SQL Engine
+            This is sql engine returns a sql engine
```
Comment (JVonderwell) [inline holoclean/dataengine.py]: says sql engine twice. just 'returns a sql engine' is fine.

File: holoclean/dataengine.py [hunk 9]
```diff
@@ -6,33 +6,33 @@
 
 class DataEngine:
     """
-    The DataEngine class which contains functionality
+    This is the class that contains functionality
```
Comment (JVonderwell) [inline holoclean/dataengine.py:9]: failing a few small pep8 tests in this file. please resolve.

**Review by JVonderwell — CHANGES_REQUESTED** [review]

**Review by gmichalo — DISMISSED** [review]

File: holoclean/dataengine.py (hunk 9 above)
Comment (ah89) [reply holoclean/dataengine.py:9]: Fixed

File: holoclean/dataengine.py (hunk 4 above)
Comment (ah89) [reply holoclean/dataengine.py:175]: Fixed

File: holoclean/dataengine.py (hunk 5 above)
Comment (ah89) [reply holoclean/dataengine.py]: Fixed

File: holoclean/dataengine.py (hunk 6 above)
Comment (ah89) [reply holoclean/dataengine.py]: fixed

File: holoclean/dataengine.py (hunk 7 above)
Comment (ah89) [reply holoclean/dataengine.py]: fixed

File: holoclean/dataengine.py (hunk 8 above)
Comment (ah89) [reply holoclean/dataengine.py]: Fixed

File: holoclean/dataset.py (hunk 3 above)
Comment (ah89) [reply holoclean/dataset.py]: Fixed

**Review by j48zheng — APPROVED** [review]

**Review by j48zheng — DISMISSED** [review]

**Review by JVonderwell — DISMISSED** [review]

File: holoclean/dataengine.py [hunk 10]
```diff
@@ -6,33 +6,33 @@
...
+    def __init__(self, holo_env):
         """
         The constructor for DataEngine class
+
         Parameters
         ----------
-        HoloEnv : HoloClean
-           This parameter is the HoloClean class from the holoclean.py
+        :param holo_env: HoloClean
+           This parameter is the HoloClean object from the holoclean.py
```
Comment (jw-mcgrath) [inline holoclean/dataengine.py]: "This parameter is [...]", is unnecessary.

File: holoclean/dataengine.py (hunk 9 above)
Comment (jw-mcgrath) [reply holoclean/dataengine.py:9]: This is the class that ...

File: holoclean/dataengine.py (hunk 6 above)
Comment (jw-mcgrath) [inline holoclean/dataengine.py]: "It returns" is redundant

File: holoclean/dataengine.py [hunk 11]
```diff
@@ -154,39 +244,51 @@ def _dataframe_to_table(self, spec_table_name, dataframe, append=0):
...
+        Parameters
+        ----------
+        :param sql_query: String
+                    The query that we want to have it results on the dataframe
+                     in the memory
+
+        Returns
+        -------
+        :return: dataframe : Dataframe
+                It returns the results of sql_query in a dataframe
```
Comment (jw-mcgrath) [inline holoclean/dataengine.py]: "It returns" is not needed

File: holoclean/dataengine.py [hunk 12]
```diff
@@ -154,39 +244,51 @@ def _dataframe_to_table(self, spec_table_name, dataframe, append=0):
...
-        Gets the schema of MySQL table
+        Getting the schema of MySQL table using "table_general_name" and
+                "dataset" to identify the table specific name
+
         Parameters
         ----------
-        dataset : DataSet
-            This parameter is the dataset object used to store the ID
+        :param dataset: DataSet
+            This parameter is a dataset object used to store the ID
```
Comment (jw-mcgrath) [inline holoclean/dataengine.py]: "This parameter is a" is redundant

File: holoclean/dataengine.py [hunk 13]
```diff
@@ -199,35 +301,47 @@ def get_schema(self, dataset, table_general_name):
...
             return df.cursor._rows[0][0]
-        except BaseException:
+        except Exception as e:
+            self.holo_env.logger.error('No such element', exc_info=e)
             return "No such element"
 
     def get_table_to_dataframe(self, table_name, dataset):
         """
-        This method get table general name and return it as spark dataframe
+        This method gets table general name and returns a spark dataframe as
```
Comment (jw-mcgrath) [inline holoclean/dataengine.py]: "This method" is not needed

File: holoclean/dataengine.py [hunk 14]
```diff
@@ -199,35 +301,47 @@ def get_schema(self, dataset, table_general_name):
...
 
-        useSpark = 1
-        return self.query(table_get, useSpark)
+        use_spark = 1
+        return self.query(table_get, use_spark)
 
     def get_db_backend(self):
-        """Return MySQL database"""
+        """
+        This method returns MySQL database
```
Comment (jw-mcgrath) [inline holoclean/dataengine.py]: "this method" is not needed.

File: holoclean/dataset.py (hunk 3 above)
Comment (jw-mcgrath) [inline holoclean/dataset.py]: "This method" is not needed

File: holoclean/dataset.py [hunk 15]
```diff
@@ -102,30 +98,88 @@ def __init__(self):
...
+        :return: dataset id : String
+        """
         fx = open('dataset_id.txt', 'w')
         fx.write(str(self.dataset_id))
         fx.close()
         return str(self.dataset_id)
 
     def return_id(self):
+        """
+        This method returns detaset id as string
```
Comment (jw-mcgrath) [inline holoclean/dataset.py]: "This method" is not needed

**Review by jw-mcgrath — DISMISSED** [review]

**Review by gmichalo — DISMISSED** [review]

**Review by gmichalo — APPROVED** [review]

**Review by thodrek — APPROVED** [review]

**Pull request merged**